### PR TITLE
pcrlock: Pad pe hash to a multiple of 8 bytes

### PR DESCRIPTION
**Pull request by daandemeyer**
All other tools (sbsigntools, osslsigncode, sbctl, goblin) do this as well so let's follow suite.

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
cc @Foxboron

**Review by Foxboron — APPROVED** [review]
lgtm

File: src/pcrlock/pehash.c [hunk 1]
```diff
@@ -135,6 +135,10 @@ int pe_hash(int fd,
                 r = hash_file(fd, mdctx, p, st.st_size - p - certificate_table->Size);
                 if (r < 0)
                         return r;
+
+                /* If the file size is not a multiple of 8 bytes, pad the hash with zero bytes. */
+                if (st.st_size % 8 != 0 && EVP_DigestUpdate(mdctx, (uint8_t[8]) {}, 8 - (st.st_size % 8)) != 1)
```
Comment (poettering) [inline src/pcrlock/pehash.c]: make the array const please. i.e. `(const uint8_t[8]) {}`

File: src/pcrlock/pehash.c [hunk 2]
```diff
@@ -135,6 +135,10 @@ int pe_hash(int fd,
                 r = hash_file(fd, mdctx, p, st.st_size - p - certificate_table->Size);
                 if (r < 0)
                         return r;
+
+                /* If the file size is not a multiple of 8 bytes, pad the hash with zero bytes. */
+                if (st.st_size % 8 != 0 && EVP_DigestUpdate(mdctx, (const uint8_t[8]) {}, 8 - (st.st_size % 8)) != 1)
```
Comment (poettering) [inline src/pcrlock/pehash.c:140]: btw, strictly speaking the array only needs to be 7 bytes long, not 8

Comment (daandemeyer) [reply src/pcrlock/pehash.c:140]: Right but that just makes the code harder to read if you ask me, a single extra byte isn't going to hurt anyone

Comment (poettering) [reply src/pcrlock/pehash.c:140]: i mean, there's a reason i didn't set the minor-suggestions label even ;-)

**Pull request merged**